core/src/main/java/com/coderanch/blackjack/Hand.java

```diff
@@ -0,0 +1,199 @@
+                .forEach((Card c) -> {
+                    var initialSize = scores.size();
+                    for (int i = 0; i < initialSize; i++) {
+                        var score = scores.get(i);
+                        scores.set(i, score + 1);
+                        if (i + 1 == initialSize) {
+                            scores.add(score + c.points());
+                        }
+                    }
+                });
+        return scores;
+    }
+
+    private static int calculateBestScore(List<Integer> possibleScores) {
+        return possibleScores.stream()
+                .sorted((o1, o2) -> Integer.compare(o2, o1))
+                .filter(i -> i <= MAX_LEGAL_SCORE)
+                .findFirst()
+                .orElse(0);
+    }
+
+    /**
+     * Gets the best score.
+     *
+     * @return the best score.
+     */
+    public int bestScore() {
+        var scores = calculateScores(this.cards);
+        return calculateBestScore(scores);
+    }
+
+    /**
+     * Issue #25.
+     *
+     * @param args main arguments
+     */
+    public static void main(String[] args) {
+        try (var iu = new InputUtility()) {
+            var miniGame = new MiniGame(iu);
+            miniGame.start();
+        }
+        catch (IOException e) {
+            System.out.println(e);
+            System.exit(1);
+        }
+    }
+
+    /**
+     * Mini game of Blackjack for issue #25.
+     */
+    private static final class MiniGame {
+
+        /**
+         * The player's current hand.
+         */
+        private Hand hand;
+
+        /**
+         * The predicate used for validating user input.
+         */
+        private Predicate<? super String> playPredicate = InputUtility.oneOfThese("hit", "pass");
+
+        /**
+         * The deck of cards used in the game.
+         */
+        private Deque<Card> deck = new ArrayDeque<>(Cards.getShuffledStandardDeck(new SecureRandom()));
+
+        /**
+         * The input utility used for getting answers.
+         */
+        private InputUtility inputUtility;
+
+        private MiniGame(InputUtility inputUtility) {
+            this.inputUtility = inputUtility;
+        }
+
+        /**
+         * Start the mini game.
+         *
+         * @throws IOException from inputUtility.
+         */
+        private void start() throws IOException {
+            hand = new Hand(deck.removeLast(), deck.removeLast());
+            while (true) {
+                displayStatus();
+                if (isGameOver()) {
+                    break;
+                }
+                var answer = inputUtility.nextString("hit or pass?", playPredicate);
+                if ("pass".equalsIgnoreCase(answer.trim())) {
+                    System.out.println("You passed.  Game over.");
+                    break;
+                }
+                dealCard();
+            }
+        }
+
+        private void displayStatus() {
+            System.out.println("Your cards are: ");
```

Comment:
Don't call `System.out` directly.

Let your game constructor take an instance of `PrintStream`.